pkg/stub/handler.go

```diff
@@ -2,96 +2,410 @@ package stub
+	ppc64CommunityContentRootDir = "/opt/openshift/arch/ppc64le/community"
+	x86                          = "x86_64"
+	ppc                          = "ppc64le"
+)
+
 func NewHandler() sdk.Handler {
-	return &Handler{}
+	h := Handler{}
+
+	h.initter = &defaultInClusterInitter{h: &h}
+	h.initter.init()
+
+	h.sdkwrapper = &defaultSDKWrapper{h: &h}
+
+	h.fileimagegetter = &defaultImageStreamFromFileGetter{h: &h}
+	h.filetemplategetter = &defaultTemplateFromFileGetter{h: &h}
+	h.filefinder = &defaultResourceFileLister{h: &h}
+
+	h.imageclientwrapper = &defaultImageStreamClientWrapper{h: &h}
+	h.templateclientwrapper = &defaultTemplateClientWrapper{h: &h}
+	h.secreteclientwrapper = &defaultSecretClientWrapper{h: &h}
+
+	h.namespace = getNamespace()
+
+	h.skippedImagestreams = make(map[string]bool)
+	h.skippedTemplates = make(map[string]bool)
+
+	return &h
 }
 
 type Handler struct {
-	// Fill me
+	initter InClusterInitter
+
+	sdkwrapper SDKWrapper
+
+	operatorCfg    *v1alpha1.SamplesResource
+	registrySecret *corev1.Secret
+
+	restconfig  *restclient.Config
+	tempclient  *templatev1client.TemplateV1Client
+	imageclient *imagev1client.ImageV1Client
+	coreclient  *corev1client.CoreV1Client
+
+	imageclientwrapper    ImageStreamClientWrapper
+	templateclientwrapper TemplateClientWrapper
+	secreteclientwrapper  SecretClientWrapper
+
+	fileimagegetter    ImageStreamFromFileGetter
+	filetemplategetter TemplateFromFileGetter
+	filefinder         ResourceFileLister
+
+	namespace string
+
+	skippedTemplates    map[string]bool
+	skippedImagestreams map[string]bool
+}
+
+func (h *Handler) manageDockerCfgSecret(deleted bool) error {
+	err := h.secreteclientwrapper.Delete("openshift", h.registrySecret.Name, &metav1.DeleteOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+	}
+	if deleted {
+		logrus.Printf("registry dockerconfig secret %s was deleted", h.registrySecret.Name)
+		h.registrySecret = nil
+		h.operatorCfg.Status.SecretCreated = false
+		h.sdkwrapper.Update()
+	} else {
+		secretToCreate := corev1.Secret{}
+		h.registrySecret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		_, err = h.secreteclientwrapper.Create("openshift", &secretToCreate)
+		if err != nil {
+			return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to create registry dockerconfig secret in openshift namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", h.operatorCfg.Spec.RegistryCredentials.Name)
+
+		h.operatorCfg.Status.SecretCreated = true
+		h.sdkwrapper.Update()
+
+	}
+
+	status := v1alpha1.ResourceCondition{
+		Type: v1alpha1.SecretUpdateComplete,
+		Time: kapis.Now(),
+	}
+	h.operatorCfg.Status.Conditions = append(h.operatorCfg.Status.Conditions, status)
```

Comment:
this is going to be an ever growing list.  I think the right way to use conditions is a fixed set of them, each having a true/false value (and a timestamp indicating when they were updated to that state).

In which case we'd want a condition like:
ImportSecretExists = true/false

I'm not sure if we care about having an "ImportSecretUpdated" condition or not, you can always refresh the "Exists" condition timestamp each time you update the secret.

Reply:
(take a look at how other resources use conditions, but the above is at least my understanding)

Reply:
Yeah, I did, and diverted per my reasoning in the other comment .... I was more interested in an audit/event log for debug purposes.  

I can certainly put a cap on the length of the array.

I still feel some resistance to conforming here, but digest my rationale, and I'll see what your response does to my resistance.

Reply:
yeah, i think if you want an audit log, that's what events are for. conditions should just reflect the current state, not the history.  And we need conditions (all operators must report conditions so we know the health of the system).  Adding event recording i'd say is bonus points (ie for now just putting that information in the operator's log output is probably sufficient)